### PR TITLE
Validate `order` kwarg in `BitArray.from_bool_array()`

### LINKED ISSUE #12673: validate `order` kwarg in `BitArray.from_bool_array()`
### Environment

- **Qiskit version**: 1.1

### What is happening?

`BitArray.from_bool_array()` has a kwarg `order` which nominally should be `'little'` or `'big'` (for bit-ordering of the input array).

The function only checks whether or not `order == 'little'`:
https://github.com/Qiskit/qiskit/blob/76af5b475b9b2a57f9eeabd3c0b793b037464630/qiskit/primitives/containers/bit_array.py#L207-L210

Would be helpful to raise an error if `order` is neither `'little'` nor `'big'`. Otherwise e.g. a user will unknowingly get the wrong result if they typo `order = 'litle'`.

### How can we reproduce the issue?

see above

### What should happen?

see above

### Any suggestions?

_No response_

### PR DESCRIPTION
**Pull request by shraddha-aangiras**
### Summary
The kwarg `order` in `BitArray.from_bool_array()` should be 'little' or 'big'. Change raises an error if order is neither of the two.
Fixes #12673

### CODE REVIEW

**Comment by t-imamichi** [comment]
Thank you for fixing the issue. Could you add a test to catch this error and a release note of bugfix?

**Comment by shraddha-aangiras** [comment]
@t-imamichi Hi, hope the test and release note added is okay. Thanks!

**Comment by t-imamichi** [comment]
Thank you for updating the PR. There seems style or lint errors.
Could you fix them following the instruction https://github.com/Qiskit/qiskit/blob/main/CONTRIBUTING.md#style-and-lint?
I guess applying black resolves one of the style error.

**Comment by shraddha-aangiras** [comment]
@t-imamichi Sorry about the formatting issue, hope this fixed it!

File: releasenotes/notes/fix-kwarg-validation-BitArray-1bf542a1fb5c15c6.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    The `kwarg` order in the function `BitArray.from_bool_array()` should be 'little' or 'big'. Added checks to raise error if invalid value is entered.
```
Comment (t-imamichi) [inline releasenotes/notes/fix-kwarg-validation-BitArray-1bf542a1fb5c15c6.yaml]:
```suggestion
    The keyword argument ``order`` of the function :meth:`.BitArray.from_bool_array` should be 'little' or 'big'. Added checks to raise error if an invalid value is entered.
```

**Comment by t-imamichi** [comment]
Thank you for fixing the style error. It looks good overall. I wrote a comment for reno.

**Review by t-imamichi — APPROVED** [review]
LGTM

**Pull request merged**